modin/backends/pandas/query_compiler.py

```diff
@@ -2473,18 +2473,26 @@ def is_reduce_fn(fn, deep_level=0):
             else:
                 internal_by = []
                 by = [by]
-        else:
+        elif by is not None:
             if not isinstance(by, list):
                 by = [by]
             internal_by = [o for o in by if hashable(o) and o in self.columns]
             internal_qc = (
                 [self.getitem_column_array(internal_by)] if len(internal_by) else []
             )
-
             by = internal_qc + by[len(internal_by) :]
 
-        broadcastable_by = [o._modin_frame for o in by if isinstance(o, type(self))]
-        not_broadcastable_by = [o for o in by if not isinstance(o, type(self))]
+        # If 'by' is None then we shouldn't move the value of 'level' to 'by', because
+        # 'level' may contain level indices. On the other hand, when 'by' is not None we
+        # must move 'level' content to 'by', since pandas does not support specifying
```

Comment:
We should not move `level ` into `by` if pandas does not support it. What is the benefit of doing this?

Reply:
answered [here](https://github.com/modin-project/modin/pull/2790/files#r584803662)